### PR TITLE
Rpc for threshold confirmations

### LINKED ISSUE #3346: Signature confirmation carries no information about likelihood of rollback
#### Problem

A signature confirmation carries no information about likelihood of rollback.  Therefore it is impossible to write stable tests that verify spend when rollbacks are possible.

#### Proposed Solution

RPC for depth and confirmation size

#3326

### PR DESCRIPTION
**Pull request by aeyakovenko**
#### Problem

Current signature status does not give enough feedback about the confirmation count.

#### Summary of Changes

Return the confirmation counter.  This is a first step, ideally we would return a `Vec<f64>` of stake confirmed percentages for each one of the forks confirming the signature.

depends on #3251 
Fixes #3346

### CODE REVIEW

File: client/src/thin_client.rs [hunk 1]
```diff
@@ -86,6 +86,36 @@ impl ThinClient {
...
+        keypair: &Keypair,
+        transaction: &mut Transaction,
+        tries: usize,
+        confs: usize,
+    ) -> io::Result<Signature> {
+        for x in 0..tries {
+            transaction.sign(&[keypair], self.get_recent_blockhash());
+            let mut buf = vec![0; transaction.serialized_size().unwrap() as usize];
+            let mut wr = std::io::Cursor::new(&mut buf[..]);
+            serialize_into(&mut wr, &transaction)
```
Comment (rob-solana) [inline client/src/thin_client.rs]: let buf = transaction.serialize()

File: client/src/thin_client.rs [hunk 2]
```diff
@@ -86,6 +86,36 @@ impl ThinClient {
...
+                .expect("serialize Transaction in pub fn transfer_signed");
+            self.transactions_socket
+                .send_to(&buf[..], &self.transactions_addr)?;
```
Comment (rob-solana) [inline client/src/thin_client.rs]: send_to(&buf, ...) ?

File: runtime/src/status_cache.rs [hunk 3]
```diff
@@ -131,13 +131,13 @@ impl<T: Clone> StatusCache<T> {
     pub fn get_signature_status_all<U>(
         checkpoints: &[U],
         signature: &Signature,
-    ) -> Option<Result<(), T>>
+    ) -> Option<(usize, Result<(), T>)>
     where
         U: Deref<Target = Self>,
     {
-        for c in checkpoints {
+        for (i, c) in checkpoints.iter().enumerate() {
```
Comment (rob-solana) [inline runtime/src/status_cache.rs]: this basically says that a child bank implies a confirmation?

Comment (aeyakovenko) [reply runtime/src/status_cache.rs]: Yea. I am going to store the stake weighted vector of parent confirmations for each bank fork.

**Comment by mvines** [comment]
Are we still reviewing here, or #3372?

**Comment by aeyakovenko** [comment]
#3372

**Pull request closed without merging**